src/app/common/StatusBadge.vue

```diff
@@ -1,21 +1,30 @@
 <template>
-  <KBadge
-    class="status"
-    :appearance="BADGE_APPEARANCE[props.status]"
-    data-testid="status-badge"
-  >
-    {{ i18n.t(`http.api.value.${props.status}`) }}
-  </KBadge>
+  <component :is="props.status === 'not_available' ? KTooltip : Anonymous">
```

Comment:
This would wrap an additional `div` around the KBadge wouldn't it? The Anonymous component doesn't add any additional HTML

Reply:
Which is better because that’s what KTooltip also does, no?

Reply:
Hmm not totally sure. I wouldn't put a `span` around a piece of text just because another piece of text has a `strong` around it.

Umm, probably an even better example from our app. I don't put a 'placeholder' HTML element in the warning cell for rows that don't have warnings in them.

Here for example, I don't wrap a div around the `t('common.collection.none')`

https://github.com/kumahq/kuma-gui/blob/765fa6b6cda93b088b4f7127c67675c1d7cd734b/src/app/zones/views/IndexView.vue#L128-L160

Reply:
Better than creating this Anonymous component it is still. This is just being pragmatic.

Reply:
Using a div here instead of this new component seems more pragmatic to me.

Reply:
But it wraps an extra div around status badges that don't need them, which I want to avoid. Whats the problem having a component that we can use like this to print nothing but the content you give it?

Reply:
It’s just simpler. But whatever, can we at least rename the component and not special case the linter rules?

Reply:
Yeah sure, what do you suggest?

Reply:
AnonymousComponent? AnonymousTemplate? Doesn’t matter much to me.

Reply:
K I changed the name of this to AnonymousComponent, lmk on that one

I also wanted to ask what is the purpose of the lint rule and making sure Components use `ThisSortOfCase` and not just `Component`?

Reply:
> I also wanted to ask what is the purpose of the lint rule and making sure Components use `ThisSortOfCase` and not just `Component`?

https://v2.vuejs.org/v2/style-guide/?redirect=true#Multi-word-component-names-essential

Can we also remove the change to the lint rule?

Reply:
Oh 🤦 , done!

---

I wonder why things like `transition` and `component` are ok? Also wondering why there is a lint rule to optionally ignore certain others 🤔